### ISSUE TITLE
Faster Horner scheme evaluation

### ISSUE THREAD

**Issue by David-Berghaus** [fredrik-johansson/arb#368]
@fredrik-johansson mentioned at the end of his paper "Faster arbitrary-precision dot product and matrix multiplication" that similar optimizations should be possible for the Horner scheme. I personally think that this would be a very useful feature to have.

**Comment by fredrik-johansson** [comment]
Yes, indeed. I will need to find a good block of time this year to write the code, it's a lot of work to do with a similar level of optimization.

**Comment by David-Berghaus** [comment]
That would be awesome, thank you! Yes, I thought about implementing it myself but it seems to be too technical for me...
